### ISSUE TITLE
Add `color-no-rgb` rule

### ISSUE THREAD

**Issue by fregante** [stylelint/stylelint#4663]
> What is the problem you're trying to solve?

Consistency, I found this code:

```css
:root {
	--background-color: #fff;
	--text-color: #202124;
	--secondary-text-color: #5f6368;
	--link-color: #3367d6;
	--input-border-color: rgba(0, 0, 0, 0.2);
}
```

The last value didn't have to use `rgba()` just to add the alpha value, it should have been

```css
:root {
	--background-color: #fff;
	--text-color: #202124;
	--secondary-text-color: #5f6368;
	--link-color: #3367d6;
	--input-border-color: #00000033;
}
```

> What solution would you like to see?

Like [color-no-hex](https://stylelint.io/user-guide/rules/color-no-hex), but for rgb/rgba instead

**Comment by jeddy3** [comment]
@fregante Thanks for the request and for using the template.

You can achieve this with either the [`function-whitelist`](https://stylelint.io/user-guide/rules/function-whitelist) or [`function-blacklist`](https://stylelint.io/user-guide/rules/function-blacklist) rules.

From [the docs on combining rules](https://stylelint.io/user-guide/rules/combine#color--and-function--rules):

> ## `color-*` and `function-*` rules

> Most `<color>` values are _functions_. As such, they can be (dis)allowed using either the `function-blacklist` or `function-whitelist` rules. Two other color representations aren't functions: named colors and hex colors. There are two specific rules that (dis)allow these: `color-named` and `color-no-hex`, respectively.

> Say you want to enforce using a named color _if one exists for your chosen color_ and use `hwb` color if one does not, e.g.:


```css
a {
  background: hwb(235, 0%, 0%); /* there is no named color equivalent for this color */
  color: black;
}
```

> If you're taking a whitelisting approach, you can do that with:

```json
{
  "color-named": "always-where-possible",
  "color-no-hex": true,
  "function-whitelist": ["hwb"]
}
```

> Or, if you're taking a blacklisting approach:

```json
{
  "color-named": "always-where-possible",
  "color-no-hex": true,
  "function-blacklist": ["/^rgb/", "/^hsl/", "gray"]
}
```

You'll probably want to use the whitelist approach as there are [a lot of different ways to represent a colour](https://drafts.csswg.org/css-color/) in CSS. For example:

```json
{
  "rules": {
    "color-no-invalid-hex": true,
    "color-named": "never",
    "function-whitelist": ["calc", "url"]
  }
}
```

This will disallow named colours, `rgb`, `rgba` and all the other functional colour representations.
